src/GoTrueApi.ts

```diff
@@ -334,4 +415,39 @@ export default class GoTrueApi {
       return { user: null, data: null, error }
     }
   }
+
+  /**
+   * Generates links to be sent via email or other.
+   * @param type The link type ("signup" or "magiclink" or "recovery" or "invite").
+   * @param email the user's email.
+   * @param password user password for signup only.
+   * @param data optional user metadata for signup only.
+   * @param redirectTo The link type ("signup" or "magiclink" or "recovery" or "invite").
+   */
+  async generateLink(
+    type: string,
```

Comment:
we should make this one an enum so it flows through to the docs

`type: 'signup' | 'magicLink' | 'recovery' | 'invite'`